### PR TITLE
🖍 [Story interactive] Color Logic for A/B/C/D Bubbles for Image Quizzes

### LINKED ISSUE #35183: [Story interactive] Use contrast color logic for the ABCD on image quizzes
When the accent color is light we should use black text color. When the accent color is dark, we should use the white text color. Can reference the getTextColorFromRGB method

### PR DESCRIPTION
**Pull request by Brandons42**
For A/B/C/D bubbles on image quizzes: Set the background color to the accent color (can be customized by user) and automatically set the text color to be a readable color based on the background color

Demo: https://interactive-img.web.app/color-logic/quiz.html

Apologies for messy commit history, only the last 5 commits are relevant (as of the initial PR)

Closes #35183 

/cc @mszylkowski 
/cc @processprocess

### CODE REVIEW

**Comment by mszylkowski** [comment]
Just a thought: IMO when the background colors is close to halfway in the brightness spectrum, the text color is easier to read when it matches the theme of the component. However, logic wise I don't think we have a way to make this work, since the contrast checker prefers the black text to the white text almost by 2x the contrast ([link](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=F34E4E)) and the logic theoretically should have nothing to do with the background of other parts of the UI that are not directly behind the text anyways.

TL;DR: There are some cases that matching the font color to the theme might make the text more readable, but I'm not sure there's much we can do since the contrast values in the example favors black text.

<table>
<tr><th>Good</th><th>Harder to read</th><th>Good</th></tr>
<tr>
<td>

</td>
<td>

</td>
<td>

</td>
</tr>
</table>

File: extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.css [hunk 1]
```diff
@@ -60,11 +47,16 @@
 
 .i-amphtml-story-interactive-post-selection
   .i-amphtml-story-interactive-img-quiz-answer-choice::before {
+  content: '' !important;
+  display: block !important;
+  width: 100% !important;
+  height: 100% !important;
+  border-radius: 50% !important;
+  position: absolute !important;
```
Comment (mszylkowski) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.css:55]: Nit: Do we need this change? I'd think it's better the other way (and also, it's not related to the ABCD coloring)

File: extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.js [hunk 2]
```diff
@@ -164,4 +175,24 @@ export class AmpStoryInteractiveImgQuiz extends AmpStoryInteractive {
...
   }
+
+  /**
+   * Set the text color of the answer choice bubble to be readable and
+   * accessible according to the background color.
+   *
+   * @param {!Element} root
+   * @private
+   */
+  setBubbleColor_(root) {
```
Comment (mszylkowski) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.js:186]: Nit: `setBubbleTextColor_` helps understand that you're just changing the text color and not the background color

**Review by mszylkowski — COMMENTED** [review]
Good use of the CSS variables, left some nits

**Comment by Brandons42** [comment]
> Just a thought: IMO when the background colors is close to halfway in the brightness spectrum, the text color is easier to read when it matches the theme of the component

I definitely agree

**Review by mszylkowski — APPROVED** [review]
